app/src/main/java/com/github/swent/echo/compose/event/EventLocationEntry.kt

```diff
@@ -1,115 +1,97 @@
 package com.github.swent.echo.compose.event
 
-import android.util.Log
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.aspectRatio
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.padding
-import androidx.compose.foundation.text.KeyboardOptions
-import androidx.compose.material.icons.Icons
-import androidx.compose.material.icons.outlined.Place
+import androidx.compose.material3.Button
 import androidx.compose.material3.Card
-import androidx.compose.material3.FilledIconButton
-import androidx.compose.material3.Icon
 import androidx.compose.material3.OutlinedButton
 import androidx.compose.material3.Text
-import androidx.compose.material3.TextField
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.stringResource
-import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.window.Dialog
 import androidx.compose.ui.window.DialogProperties
 import com.github.swent.echo.R
+import com.github.swent.echo.compose.map.LocationDisplayer
+import com.github.swent.echo.compose.map.LocationSelector
 import com.github.swent.echo.data.model.Location
 
 /** this composable contains the Location title, text field and button */
 @Composable
-fun EventLocationEntry(location: Location, onLocationChanged: (newLocation: Location) -> Unit) {
-    var showPickLocationDialog by remember { mutableStateOf(false) }
-    if (showPickLocationDialog) {
+fun EventLocationEntry(location: Location, onLocationChanged: (Location) -> Unit) {
+    var editLocation by remember { mutableStateOf(false) }
+    var displayedLocation by remember { mutableStateOf(location) }
+    if (editLocation) {
         SelectLocationDialog(
             currentLocation = location,
-            onDismissRequest = { showPickLocationDialog = false },
-            onSelectLocation = onLocationChanged
-        )
+            onDismissRequest = { editLocation = false }
+        ) {
+            displayedLocation = it
+            onLocationChanged(it)
+        }
     }
-    Row(modifier = Modifier.fillMaxWidth(1f)) {
+    Column(modifier = Modifier.fillMaxWidth(1f)) {
         EventTextEntry(
             name = stringResource(R.string.edit_event_screen_location),
             value = location.name,
             modifier = Modifier.fillMaxWidth(0.8f)
         ) {
```

Comment:
Here I'd remove the `0.8f` as there is no more a marker icon to open the location dialog. Makes it more consistent with the rest...